### PR TITLE
Add spm support

### PR DESCRIPTION
**Pull request by tahirmt**
SPM does not support mixing Objective-C and Swift in one target. To get around this limitation we now define multiple targets that are swift and objc exclusive. `Nimble-Snapshots-Swift-Core` is needed so we can expose `CurrentTestCaseTracker` to Objective-C as we need to register it in the `load` function which isn't supported by swift. `Nimble-Snapshots-Core` contains all the Objective-C code importing the swift core framework. 

Finally, the product `Nimble_Snapshots` imports both of these and creates one binary that the consumers can use. 
We also use `@_exported import` for the internal frameworks so consumers don't need to import them to use the code.

### CODE REVIEW

File: Tests/Nimble-SnapshotsTests/Nimble_SnapshotsTests.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import XCTest
+@testable import Nimble_Snapshots
+
+final class Nimble_SnapshotsTests: XCTestCase {
+    func testExample() {
+        // This is an example of a functional test case.
+        // Use XCTAssert and related functions to verify your tests produce the correct
+        // results.
+        XCTAssertEqual(Nimble_Snapshots().text, "Hello, World!")
```
Comment (ketenshi) [inline Tests/Nimble-SnapshotsTests/Nimble_SnapshotsTests.swift]: How does this work? I can't seem to find where "Hello, World!" is set in the code

File: Sources/Nimble_Snapshots/HaveValidSnapshot.swift [hunk 2]
```diff
@@ -44,10 +47,10 @@ public class FBSnapshotTest: NSObject {
                                referenceDirectory: String,
                                tolerance: CGFloat,
                                perPixelTolerance: CGFloat,
-                               filename: String,
+                               filename: FileString,
```
Comment (ketenshi) [inline Sources/Nimble_Snapshots/HaveValidSnapshot.swift:50]: What's the reason for using `FileString` instead?

File: Sources/Nimble_Snapshots/FileString+String.swift [hunk 3]
```diff
@@ -0,0 +1,15 @@
+//
+//  FileString+String.swift
+//  
+//
+//  Created by Mahmood Tahir on 2021-06-10.
+//
+
+import Foundation
+import Nimble
```
Comment (ketenshi) [inline Sources/Nimble_Snapshots/FileString+String.swift:9]: Can probably omit importing Nimble

**Comment by ketenshi** [comment]
Can you speak to why Swift-Core and Core are needed to get SPM working?

File: Tests/Nimble-SnapshotsTests/Nimble_SnapshotsTests.swift (hunk 1 above)
Comment (tahirmt) [reply Tests/Nimble-SnapshotsTests/Nimble_SnapshotsTests.swift]: Oh I should remove this test target. The tests are currently done in the bootstrap project

File: Sources/Nimble_Snapshots/HaveValidSnapshot.swift (hunk 2 above)
Comment (tahirmt) [reply Sources/Nimble_Snapshots/HaveValidSnapshot.swift:50]: `FileString` is defined by `Nimble`. It's either `StaticString` for swift package or `String` so I had to adopt it

File: Sources/Nimble_Snapshots/FileString+String.swift (hunk 3 above)
Comment (tahirmt) [reply Sources/Nimble_Snapshots/FileString+String.swift:9]: `Nimble` defines `FileString`

File: Tests/Nimble-SnapshotsTests/Nimble_SnapshotsTests.swift (hunk 1 above)
Comment (tahirmt) [reply Tests/Nimble-SnapshotsTests/Nimble_SnapshotsTests.swift]: Removed

**Comment by tahirmt** [comment]
> Can you speak to why Swift-Core and Core are needed to get SPM working?

@ketenshi added a description to the PR explaining this.

**Review by ketenshi — APPROVED** [review]
Changes LGTM

I can't resolve convos for some reason

**Pull request merged**